server/models.py

```diff
@@ -306,6 +308,8 @@ def can(cls, obj, user, action):
         is_staff = user.is_enrolled(obj.course.id, STAFF_ROLES)
         if action == "view":
             return is_staff or obj.visible
+        if action == "publish":
```

Comment:
Nit: `publish` is kind of confusing since there is also a publish action on assignments. Maybe we could add a comment or change the name to publish-scores 

Also: Not being able to publish scores for a hidden assignment seems a bit off? (For example - a unpublished assignment might want to default to having scores visible) 

Not a big deal - just wondering.